internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -7,26 +7,150 @@ import (
 	"testing"
 	"time"
 
-	"github.com/go-logr/logr"
 	ngxclient "github.com/nginxinc/nginx-plus-go-client/client"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime/runtimefakes"
 )
 
 var _ = Describe("NGINX Runtime Manager", func() {
 	It("returns whether or not we're using NGINX Plus", func() {
-		mgr := NewManagerImpl(nil, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr := runtime.NewManagerImpl(nil, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeFalse())
 
-		mgr = NewManagerImpl(&ngxclient.NginxClient{}, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr = runtime.NewManagerImpl(&ngxclient.NginxClient{}, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeTrue())
 	})
+
+	var (
+		err             error
+		manager         runtime.Manager
+		upstreamServers []ngxclient.UpstreamServer
+		ngxPlusClient   *runtimefakes.FakeNginxPlusClient
+		process         *runtimefakes.FakeProcessHandler
+
+		metrics      *runtimefakes.FakeMetricsCollector
+		verifyClient *runtimefakes.FakeVerifyClient
+	)
+
+	BeforeEach(func() {
+		upstreamServers = []ngxclient.UpstreamServer{
+			{},
+		}
+	})
+
+	Context("Reload", func() {
+		BeforeEach(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			process = &runtimefakes.FakeProcessHandler{}
+			metrics = &runtimefakes.FakeMetricsCollector{}
+			verifyClient = &runtimefakes.FakeVerifyClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+		})
+
+		When("MetricsCollector is nil", func() {
```

Comment:
Since this is the successful/normal case, I would omit the When block, and just have:

```
It("Is successful")
```

When combined with the descriptions in the Ginkgo containers that contain this test you will get:

NGINX Runtime Manager
Reload
Is successful

Reply:
make sense, will do